File: leetcode/Go/trees/104/104.go
package main

type TreeNode struct {
	Val   int
	Left  *TreeNode
	Right *TreeNode
}

func traverse(root *TreeNode, level int, maxLevel *int) {
	if root == nil {
		return
	}
	traverse(root.Left, level+1, maxLevel)
	if level > *maxLevel {
		*maxLevel = level
	}
	traverse(root.Right, level+1, maxLevel)
}

func maxDepth(root *TreeNode) int {
	maxLevel := 0
	traverse(root, 1, &maxLevel)
	return maxLevel
}

File: leetcode/Python/trees/104.py
# Definition for a binary tree node.
from typing import Optional


class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right
class Solution:
    def maxDepth(self, root: Optional[TreeNode]) -> int:
        def traverseTree(root, depth: int) -> int:
            if root is None:
                return depth
            return max(traverseTree(root.left, depth+1), traverseTree(root.right, depth+1))
        return traverseTree(root, 0)
        
